include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -79,15 +78,43 @@ namespace internal
                               const FiniteElement<dim> &fe_in,
                               const unsigned int        base_element_number)
     {
-      this->quadrature             = quad;
       const FiniteElement<dim> *fe = &fe_in.base_element(base_element_number);
+      n_dimensions                 = dim;
+      n_components                 = fe_in.n_components();
 
       Assert(fe->n_components() == 1,
              ExcMessage("FEEvaluation only works for scalar finite elements."));
 
-      fe_degree     = fe->degree;
-      n_q_points_1d = quad.size();
-
+      // assuming isotropy of dimensions and components
```

Comment:
Can you elaborate how the extension to anisotropy will look like? Also have you thought about having different quadrature rules in different directions?

Reply:
In the future `reinit()` is replaced by an interface to `FiniteElement` (as suggested here #9544). Using different quadrature rules simply extends the set of `UnivariateShapeData`s stored.